README.md
# wybaby.github.io

book/history
14:30:26  113.208.115.178    0  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Mobile/14F89 MicroMessenger/6.5.12 NetType/WIFI Language/zh_CN
07:27:44  113.208.115.190  高效能  2  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
07:34:16  113.208.115.190  高效能  2  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
07:43:47  113.208.115.190  crontab  0  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
06:05:27  221.221.208.17  善良天使  2  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Version/10.0 Mobile/14F89 Safari/602.1
07:34:22  113.208.115.190  人性中的善良天使  2  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
07:36:29  113.208.115.190  人  27  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
16:36:17  221.221.208.17  人类行为  0  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Version/10.0 Mobile/14F89 Safari/602.1
16:41:35  221.221.208.17  伦理学  1  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Version/10.0 Mobile/14F89 Safari/602.1
10:06:14  113.208.115.190  人  27  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 10:08:26  113.208.115.190  测试  4  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 12:18:55  113.208.115.190  论人类不平等的起源和基础  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 13:55:07  113.208.115.190  证券市场  0  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 13:55:30  113.208.115.190  基础知识  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 14:55:01  166.111.139.47  逻辑学  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 16:31:27  113.208.115.190  五大问题  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 20:23:27  221.221.208.17  有机朗肯  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 20:23:47  221.221.208.17  有机朗肯  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 20:23:53  221.221.208.17  有机朗肯  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 20:23:57  221.221.208.17  幸福之路  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 20:24:01  221.221.208.17  有机朗肯  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 20:24:24  221.221.208.17  有机朗肯  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170731 20:26:54  221.221.208.17  高效能  3  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170801 09:01:31  113.208.115.190  逻辑  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170802 10:26:30  113.208.115.190  力学  6  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170803 06:13:47  221.221.208.17  当我们  0  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.0 Mobile/14G60 Safari/602.1
20170803 06:14:02  221.221.208.17  谈论  0  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.0 Mobile/14G60 Safari/602.1
20170805 06:57:21  221.221.208.17  故事  3  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.0 Mobile/14G60 Safari/602.1
20170806 20:41:29  221.221.208.17  流体  2  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
20170808 07:59:52  113.208.115.190  功利  1  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.0 Mobile/14G60 Safari/602.1
20170809 06:06:23  111.194.94.119  功利  1  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.0 Mobile/14G60 Safari/602.1
20170809 09:13:33  104.236.177.93  功利主义  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36
20170810 08:02:40  61.148.243.115  曾国藩  4  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.0 Mobile/14G60 Safari/602.1
20170810 08:03:00  61.148.243.115  曾  6  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.0 Mobile/14G60 Safari/602.1
20170811 09:04:24  113.208.115.190  热工基础  1  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36
20170812 06:07:16  111.194.94.119  高效  3  Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.0 Mobile/14G60 Safari/602.1

js/bookquery.js
var res = {};

function loading() {
Papa.parse("https://wybaby.github.io/book/book.txt", {
	download: true,
	complete: function(results) {
		//console.log(results);
		var $tblSts = "<table border='0'>";
		var $no = 0;
		var $update = "";
		for (var i=0; i<results.data.length; ++i) {
			var b = results.data[i];
			if (b[0].length>0) {
				$no = b[0];
			};
			if (b[0].length==0 && (typeof(b[1])!="undefined") && b[1].indexOf("2017")>=0) {
				$update = b[1];
			};
		}
		$tblSts += "<tr><td>更新日期 " + $update + "</td><td></td><td>" + "共 " + $no + " 本" + "</td></tr>";
		$tblSts += "</table>";
		$("#stsTbl").html($tblSts);
		res = results;
	}
});

}

function bquery() {
    var $searchname = trimStr(document.getElementById('searched_content').value);
    if ($searchname.length<=0) {
        return;
    }
/*
Papa.parse("https://wybaby.github.io/book/book.txt", {
	download: true,
	complete: function(results) {
		*/
		//console.log(res);
        var $found = 0;
        var $tblStr = "";//<table border='0'>";
        for (var i=0; i<res.data.length; ++i) {
            var b = res.data[i];
            if (b.length<20) {
                continue;
            }
//            console.log(b[3]);
            if (b[3].indexOf($searchname)>=0) {
                $found++;
				$tblStr += "<tr><td class='table-x'>" + "序号" + "</td><td>" + b[0] + "</td></tr>";
                $tblStr += "<tr><td class='table-x'>" + "编号" + "</td><td>" + b[1] + "</td></tr>";
                $tblStr += "<tr><td class='table-x'>" + "书名" + "</td><td>" + b[3] 				
				if (b[4].length>0) {
					$tblStr += (" - " + b[4]);
				};
				$tblStr += "</td></tr>";
                $tblStr += "<tr><td class='table-x'>" + "作者" + "</td><td>" + b[5] + "</td></tr>";
                $tblStr += "<tr><td class='table-x'>" + "ISBN" + "</td><td>" + "<a target='_blank' href='https://book.douban.com/subject_search?search_text=" +b[13] + "'>" + b[13] + "</a>" + "</td></tr>";
                $tblStr += "<tr><td class='table-x'>" + "状态" + "</td><td>" + b[2] + "</td></tr>";
                $tblStr += "<tr><td class='table-x'>" + "出版日期" + "</td><td>" + b[7] + "</td></tr>";
                $tblStr += "<tr><td class='table-x'>" + "购买日期" + "</td><td>" + b[16] + "</td></tr>";
		if (b[17].length>0) {
			$tblStr += "<tr><td class='table-x'>" + "阅读日期" + "</td><td>" + b[17] + "</td></tr>";
		};
		if (b[18].length>0) {
			$tblStr += "<tr><td class='table-x'>" + "" + "</td><td>" + b[18] + "</td></tr>";
		};
		if (b[19].length>0) {
			$tblStr += "<tr><td class='table-x'>" + "打分" + "</td><td>" + b[19] + "</td></tr>";
		};
		if (b[20].length>0) {
			$tblStr += "<tr><td class='table-x'>" + "简评" + "</td><td>" + b[20] + "</td></tr>";
		};
		if (b[21].length>0) {
			$tblStr += "<tr><td class='table-x'>" + "江湖地位" + "</td><td>" + b[21] + "</td></tr>";
		};
                $tblStr += "<tr><td>----------</td></tr>";
            }
        }
		$tblStr += "</table>";
        if ($found==0) {
            $("#resTbl").html("<table><tr></tr><tr><td>没买过</td></tr></table>");
        } else {
			$tblStr = "<table border='0'>" + "<tr><td>" + "【找到 " + $found + " 本】" + "</td></tr>" + $tblStr;
            $("#resTbl").html($tblStr);
        }
//	}
//});

}

function trimStr(str){return str.replace(/(^\s*)|(\s*$)/g,"");}


book/getfiletail.php
<?php

/**
 * Slightly modified version of http://www.geekality.net/2011/05/28/php-tail-tackling-large-files/
 * @author Torleif Berger, Lorenzo Stanco
 * @link http://stackoverflow.com/a/15025877/995958
 * @github https://gist.github.com/lorenzos/1711e81a9162320fde20
 * @license http://creativecommons.org/licenses/by/3.0/
 */
function tailCustom($filepath, $lines = 1, $adaptive = true) {
    // Open file
    $f = @fopen($filepath, "rb");
    if ($f === false) return false;
    // Sets buffer size, according to the number of lines to retrieve.
    // This gives a performance boost when reading a few lines from the file.
    if (!$adaptive) $buffer = 4096;
    else $buffer = ($lines < 2 ? 64 : ($lines < 10 ? 512 : 4096));
    // Jump to last character
    fseek($f, -1, SEEK_END);
    // Read it and adjust line number if necessary
    // (Otherwise the result would be wrong if file doesn't end with a blank line)
    if (fread($f, 1) != "\n") $lines -= 1;

    // Start reading
    $output = '';
    $chunk = '';
    // While we would like more
    while (ftell($f) > 0 && $lines >= 0) {
        // Figure out how far back we should jump
        $seek = min(ftell($f), $buffer);
        // Do the jump (backwards, relative to where we are)
        fseek($f, -$seek, SEEK_CUR);
        // Read a chunk and prepend it to our output
        $output = ($chunk = fread($f, $seek)) . $output;
        // Jump back to where we started reading
        fseek($f, -mb_strlen($chunk, '8bit'), SEEK_CUR);
        // Decrease our line counter
        $lines -= substr_count($chunk, "\n");
    }
    // While we have too many lines
    // (Because of buffer size we might have read too many)
    while ($lines++ < 0) {
        // Find first newline and remove all text before that
        $output = substr($output, strpos($output, "\n") + 1);
    }
    // Close file and return
    fclose($f);
    return trim($output);
}
?>



book/index.php
<html>
    <head>
        <link rel='icon' href='book.ico'>
        <link rel='apple-touch-icon' href='iphone_icon.png'>
        <title>买没买过</title>
        <link rel='stylesheet' type='text/css' href='book.css'>
        <script type="text/javascript">
            function FocusOnInput() {
                document.getElementById("searched_content").focus();
            }
        </script>
    </head>
    <body onload=FocusOnInput()>
<?php
date_default_timezone_set("Asia/Shanghai");
/*
echo "<html>" . "\n";
echo "<head>" . "\n";
echo "<link href='book.ico' rel='shortcut ico'>" . "\n";
echo "<link rel='apple-touch-icon' href='iphone_icon.png'>" . "\n";
echo "<title>买没买过</title>" . "\n";
echo "<link rel='stylesheet' type='text/css' href='book.css'>" . "\n";
echo "</head>" . "\n";
*/


$filename = "book.txt";
$update_time = date("Y/m/d H:i", filemtime($filename));
$fp = fopen($filename, "r");
$last_num = false;
$current_num = false;
$total = 0;
$no_read = 0;
//$found_rows[];
while ( $line = fgets($fp) ) {
    $row = explode(",", $line);
    $last_num = $current_num;
    $current_num = is_numeric($row[0]);
    if ($current_num && $last_num) {
        $total = intval($row[0]);
        if ((strval($row[2]) != "未读") && (strval($row[2]) != "阅读中")) {
            $no_read+=1;
        }
    }
}
fclose($fp);
$status = "更新时间 " . $update_time . " 共 " . $total . " 本 已读 ". $no_read . " 本 (" . round($no_read/$total*100, 1) . "%)";
?>
        <form method="post" action="<?php echo $_SERVER["PHP_SELF"]; ?>">
            <div class='table-b'><table border='0'><tr><td><?php echo $status;?></td></tr></table></div>
            <input type='text' name='bookname' value='<?php if(isset($_POST['bookname'])){echo trim($_POST['bookname']);} ?>' id='searched_content' title='序号、书名或作者' onfocus="this.select()" onmouseover="this.select()"/>
            <input type='submit' name='submit' value='Go' id='search' title='gogogo' />
            <input type='submit' name='random' value='偶遇' id='random' />
        </form>
<?php
/*
echo "<div class='table-b'><table border='0'><tr><td>". $status . "</td></tr></table>" . "\n";
echo "<input type='text' name='bookname'  id='searched_content' title='书名' />" . "\n";
echo "<input type='submit' value='Go' class='button' id='search' title='gogogo' />" . "\n";
*/
if ($_SERVER["REQUEST_METHOD"] == "POST") { 
    $search_name = trim($_POST["bookname"]);
    if (strlen($search_name) < 1 && $_REQUEST['submit']) exit;
    $fp = fopen($filename, "r");
    $tblStr = "";
    $line = fgets($fp); // 抬头一行不参与
    if ($_REQUEST['submit']) {
        while ( $line = fgets($fp) ) {
            $row = explode(",", $line);
            $pos1 = strpos(strtolower($row[0]), strtolower($search_name));  // 编号
            $pos2 = strpos(strtolower($row[3]), strtolower($search_name));  // 名称
            $pos3 = strpos(strtolower($row[4]), strtolower($search_name));  // 副标题
            $pos4 = strpos(strtolower($row[5]), strtolower($search_name));  // 作者
            if ($pos1===false && $pos2===false && $pos3===false && $pos4===false) {
            } else {
                $found_rows[] = $row;
            }
        }
        $fpHistory = fopen("history", "a");
        fwrite($fpHistory, date("Ymd H:i:s") . "  " . $_SERVER["REMOTE_ADDR"] . "  " . trim($search_name) . "  " . strval(count($found_rows)) . "  " . $_SERVER['HTTP_USER_AGENT'] . "\n");
        fclose($fpHistory);
    } elseif ($_REQUEST['random']) {
        $r = rand(1, $total);
        $i = 0;
        while(++$i <= $r) {
            $row = explode(",", fgets($fp));
        }
        $found_rows[] = $row;
    }
    fclose($fp);
    $Tab4 = "    ";
    $Tab8 = $Tab4.$Tab4;
    $Tab12 = $Tab4.$Tab8;
    if (count($found_rows) > 0) {
        foreach ($found_rows as $row) {
            for ($i=0; $i<count($row); ++$i) {
                $row[$i] = trim($row[$i]);
            }
            $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "序号" . "</td><td>" . $row[0] . "</td></tr>" . "\n";
            $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "编号" . "</td><td>" . $row[1] . "</td></tr>" . "\n";
            $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "书名" . "</td><td>" . $row[3];
            if (strlen($row[4])>0) {
                $tblStr .= (" - " . $row[4]);
            };
            if (strpos($row[16],"电子书")===false) {
            } else {$tblStr .= " (电子书)";}
            $tblStr .= "</td></tr>" . "\n";
            $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "作者" . "</td><td>" . $row[5] . "</td></tr>" . "\n";
            // 增加column“写作年份”，8+下标全部加1
            // $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "ISBN" . "</td><td>" . "<a title='豆瓣' target='_blank' href='https://book.douban.com/subject_search?search_text=" . $row[14] . "'>" . $row[14] . "</a>" . "</td></tr>" . "\n";
            // 豆瓣直接改为isbn跳到书的页面
            $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "ISBN" . "</td><td>" . "<a title='豆瓣' target='_blank' href='https://book.douban.com/isbn/" . $row[14] . "'>" . $row[14] . "</a>" . "</td></tr>" . "\n";
            $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "状态" . "</td><td>" . $row[2] . "</td></tr>" . "\n";
            $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "出版日期" . "</td><td>" . $row[7] . " (" . $row[8] . "年) </td></tr>" . "\n";
            $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "购买日期" . "</td><td>" . $row[17] . "</td></tr>" . "\n";
            if (strlen($row[18])>0) {
                $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "阅读日期" . "</td><td>" . $row[18] . "</td></tr>" . "\n";
            };
            if (strlen($row[19])>0) {
                $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "" . "</td><td>" . $row[19] . "</td></tr>" . "\n";
            };
            if (strlen($row[20])>0) {
                $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "打分" . "</td><td>" . $row[20] . "</td></tr>" . "\n";
            };
            if (strlen($row[21])>0) {
                $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "简评" . "</td><td>" . $row[21] . "</td></tr>" . "\n";
            };
            if (strlen($row[22])>0) {
                $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "江湖地位" . "</td><td>" . $row[22] . "</td></tr>" . "\n";
            };
            // $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "索书号" . "</td><td>" . "<a target='_blank' href='http://innopac.lib.tsinghua.edu.cn/search*chx/c?SEARCH=" . $row[23] . "'>" . $row[23] . "</a>" . "</td></tr>" . "\n";
            // 图书馆更新了检索系统，11年来第一次，更新了url
            $tblStr .= $Tab12 . "<tr><td class='table-x'>" . "索书号" . "</td><td>" . "<a title='清华LIB' target='_blank' href='https://tsinghua-primo.hosted.exlibrisgroup.com/primo-explore/search?query=any,contains," . $row[23] . "&tab=print_tab&search_scope=print_scope&vid=86THU&lang=zh_CN&offset=0'>" . $row[23] . "</a>" . "</td></tr>" . "\n";

            $tblStr .= $Tab12 . "<tr><td>------------</td></tr>" . "\n";
        }
        $tblStr .= $Tab8 . "</table></div>" . "\n";
        if ($_REQUEST['submit']) {
            $tblStr = "<tr><td colspan=2>" . "【找到 " . count($found_rows) . " 本 \"". $search_name . "\"】" . "</td></tr>" . "\n" . $tblStr;
        } elseif ($_REQUEST['random']) {
            $tblStr = "<tr><td colspan=2>【手气不错】</td></tr>" . "\n" . $tblStr;
        }
        $tblStr = $Tab8 . "<div class='table-b'><table border='0'>" . "\n" . $Tab12 . $tblStr;
    } else {
        $tblStr = $Tab8 . "<div class='table-b'><table><tr></tr><tr><td>没买过 \"" . $search_name . "\" </td></tr></table></div>" . "\n";
    }
    echo $tblStr;
}
?>
    </body>
<html>
